Bluetooth: Fixes for shell sample

Vudentz opened the pull request:
Some leftovers of the port to the new shell.

Vudentz committed:
Bluetooth: shell: Fix using \n with print/error

Those macros already contain line termination.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

Vudentz committed:
Bluetooth: samples: shell: Don't use printk

Make use of shell_fprintf whenever possible.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

jarz-nordic commented:
@Vudentz : Would you mind to rebase it to #10322 ?

jarz-nordic commented on tests/bluetooth/shell/src/main.c:
```diff
@@ -86,21 +91,20 @@ static void cmd_hrs(const struct shell *shell, size_t argc, char **argv)
 
-	shell_fprintf(shell, SHELL_ERROR, "%s:%s%s\r\n", argv[0],
-		      "unknown parameter: ", argv[1]);
+	error(shell, "%s:%s%s", argv[0], "unknown parameter: ", argv[1]);
```
to speed up code execution you could rephrase it like that:

`error(shell, "%s:%unknown parameter: %s", argv[0], argv[1]);`

Vudentz replied:
Yep, I was wondering about this when doing the initial code but then I forgot to mention why it was done like that in first place.

Vudentz commented:
@jarz-nordic done

Vudentz committed:
Bluetooth: Shell: Make error message for unknown parameter consistent

Also simplify it by having the 'unknown parameter:' as part of the
message format.

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

carlescufi commented:
@Vudentz CI failing
